projects/common/src/time/page-time-range-preference.service.ts

```diff
@@ -1,53 +1,74 @@
 import { Injectable } from '@angular/core';
 import { isNil } from 'lodash-es';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map, shareReplay, take } from 'rxjs/operators';
+import { ApplicationFeature } from '../constants/application-constants';
+import { FeatureStateResolver } from '../feature/state/feature-state.resolver';
+import { FeatureState } from '../feature/state/feature.state';
 import { NavigationService } from '../navigation/navigation.service';
 import { PreferenceService, StorageType } from '../preference/preference.service';
+import { RelativeTimeRange } from './relative-time-range';
+import { TimeDuration } from './time-duration';
 import { TimeRange } from './time-range';
 import { TimeRangeService } from './time-range.service';
+import { TimeUnit } from './time-unit.type';
 
 @Injectable({ providedIn: 'root' })
 export class PageTimeRangePreferenceService {
   private static readonly STORAGE_TYPE: StorageType = StorageType.Local;
   private static readonly TIME_RANGE_PREFERENCE_KEY: string = 'page-time-range';
 
+  private readonly globalDefaultTimeRange: TimeRange = new RelativeTimeRange(new TimeDuration(30, TimeUnit.Minute));
+
   private readonly pageTimeRangeStringDictionary$: Observable<PageTimeRangeStringDictionary>;
 
   public constructor(
     private readonly preferenceService: PreferenceService,
     private readonly timeRangeService: TimeRangeService,
-    private readonly navigationService: NavigationService
+    private readonly navigationService: NavigationService,
+    private readonly featureStateResolver: FeatureStateResolver
   ) {
     this.pageTimeRangeStringDictionary$ = this.buildPageTimeRangeObservable();
   }
 
-  public getTimeRangePreferenceForPage(path: string): Observable<TimeRange | undefined> {
-    return this.pageTimeRangeStringDictionary$.pipe(
-      map(pageTimeRangeStringDictionary => {
-        if (isNil(pageTimeRangeStringDictionary[path])) {
-          return this.getDefaultPageTimeRange(path);
+  public getTimeRangePreferenceForPage(rootLevelPath: string): Observable<TimeRange> {
+    return combineLatest([
+      this.pageTimeRangeStringDictionary$,
+      this.featureStateResolver.getCombinedFeatureState([
+        ApplicationFeature.PageTimeRange,
+        ApplicationFeature.NavigationRedesign
+      ])
+    ]).pipe(
+      map(([pageTimeRangeStringDictionary, featureState]) => {
+        if (featureState === FeatureState.Enabled) {
+          if (isNil(pageTimeRangeStringDictionary[rootLevelPath])) {
+            // Right side for when FFs are enabled but page path doesn't have 'defaultTimeRange' set on AR data
+            return this.getDefaultPageTimeRange() ?? this.globalDefaultTimeRange;
```

Comment:
nit: getDefaultPageTimeRange without arguments reads more global than it is - maybe getDefaultTimeRangeForCurrentRoute ?